TangoRosStreamer/app/src/main/java/eu/intermodalics/tango_ros_streamer/RunningActivity.java

```diff
@@ -159,6 +162,24 @@ public void onTangoRosErrorHook(int returnCode) {
         }
     }
 
+    @Override
+    public void onTangoRosStartHook() {
+        // Tango ROS node thread started, so Tango Connect service will be advertised.
+        // Ready to call service to connect to Tango once service is advertised.
+        // ROS Java does not provide a way to wait until service becomes available so retry to
+        // connect 20 times.
+        int count = 0;
+        while (count < 20 && !mTangoServiceClientNode.callTangoConnectService(TangoConnectRequest.CONNECT)) {
+            try {
+                count++;
+                Log.e(TAG, "Trying to connect to Tango, attempt " + count);
+                Thread.sleep(200);
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+        }
```

Comment:
@PerrineAguiar Do you know a way to know when the `execute` hook from a node is finished?